daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Result.scala

```diff
@@ -90,6 +98,45 @@ final case class ResultNeedKey[A](
     resume: Option[ContractId] => Result[A],
 ) extends Result[A]
 
+/** Whether a given contract can be fetched by key, i.e., actAs union readAs
+  *    contains at least one stakeholder.
+  */
+sealed trait VisibleByKey {
```

Comment:
Isn't the compiler complaining that this is not `Serializable with Product`?

Reply:
Ah sorry, only when inferring it from its children it is

Reply:
Happy to add it if you prefer, I don’t have strong feelings here either way.

Reply:
I think it's bloat usually, so if the compiler doesn't want it, it's fine